feat(KL-162/검색 결과 표시): add search result to preset for FeedPage

seoulyego opened the pull request:
## 📌 연관된 이슈
[KL-162/검색 결과 표시](https://ohhamma.atlassian.net/browse/KL-162)

## 📝 작업 내용
- data의 property명을 변경하였습니다.
- product data의 fetching을 컴포넌트에서 훅으로 변경하였습니다
- 페이지 버그를 수정하였습니다.
  - 필터가 설정될 때마다 요청할 페이지 번호를 0으로 초기화합니다.
- 검색 결과를 반영한 필터 설정
  - ReviewPage에서 검색 데이터를 state로 전달합니다.
  - location.state를 확인하여 상위 데이터를 찾아 함께 select 리스트에 추가합니다.
- 자잘한 수정 (변수명, 디자인, 버그 등)

## 🌳 작업 브랜치명
KL-162/SearchResult

## 📸 스크린샷 (선택)

## 💬 리뷰 요구사항 (선택)
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요

[KL-162]: https://ohhamma.atlassian.net/browse/KL-162?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
	- Enhanced visual representation of product information with improved separators.
	- Improved handling of category and subcategory selections based on routing state.
	- Introduced a new loader for fetching hierarchical data related to regions and categories.
	- Added a custom hook for managing product data fetching, simplifying component interfaces.
	- Implemented a new state management method for more granular control over selected regions.
	- Added a new component for managing and displaying subcategories within filters.

- **Bug Fixes**
	- Corrected conditional rendering logic in the CitySection component to prevent display errors.

- **Refactor**
	- Simplified component structures and state management across several components, including the transition to a custom hook for product data fetching.
	- Streamlined pagination handling and data fetching logic for improved performance and clarity.
	- Updated the layout structure of components for better organization.
	- Standardized naming conventions for consistency and readability in the codebase.
	- Refactored tag filtering components to enhance user feedback and simplify data handling.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

seoulyego committed:
KL-162/feat: use location state

FeedPage의 location state를 확인합니다.

seoulyego committed:
KL-162/refactor: change a component to a hook

ProductDataFetcher 컴포넌트를 useProductData 훅으로 변경하였습니다.

seoulyego committed:
KL-162/fix: fix page bug

페이지 버그를 고쳤습니다.
- 필터 추가 시 요청할 페이지 번호를 0으로 추가합니다.

seoulyego committed:
KL-162/style: change separator

국가명과 카테고리명 사이의 구분점을 변경하였습니다.

seoulyego committed:
KL-162/fix: fix typos in property names in fetched data

fetching된 category의 subcategories의 오타를 수정하였습니다.

seoulyego committed:
KL-162/style: rename parameter

setSelectedCountry의 parameter 이름을 변경하였습니다.

seoulyego committed:
KL-162/style: remove console.log on FeedPage

FeedPage의 location.state의 테스트용 출력 코드를 제거하였습니다.

seoulyego committed:
KL-162/feat: pass search data to link state on ReviewPage

리뷰 상세 페이지에서 검색 데이터를 link state로 전달합니다.

seoulyego committed:
KL-162/feat: preset search result for filters on FeedPage

필터에 검색 결과를 select합니다.
- lacation.state를 통해 검색 결과 데이터를 전달합니다.
- 전달된 데이터를 확인하여 상위의 데이터를 찾습니다.
- 설정할 데이터를 select 리스트에 추가합니다.

sayoonnn reviewed (dismissed):
LGTM

seoulyego committed:
KL-162/design: change the margin-bottom to row-gap

StyledList의 margin-bottom 추가 분을 FeedContainer row-gap에 반영하였습니다.

seoulyego committed:
KL-162/style: minor changes

- PreviewContent의 좋아요 버튼의 크기를 조정하였습니다.
- ProductList의 PropTypes를 간소화 하였습니다.

seoulyego committed:
KL-162/feat: add loader for FeedPage

FeedPage에 loader를 추가하였습니다.
- 필요한 fetching을 loader에서 수행합니다.

seoulyego committed:
KL-162/refactor: apply debouce fetch

- response의 순서가 간헐적으로 바뀌는 문제를 해결하였습니다.
- useProductQuery를 필터 별로 구분하였습니다.
- parameter query의 구성이 변경되었습니다.

seoulyego committed:
KL-162/refactor: change the preset of the BasicFilter to FeedPage

- BasicFilter의 사전 설정을 FeedPage로 이동하였습니다.
- 각 Section의 PropTypes를 간소화하였습니다.

seoulyego committed:
KL-162/refactor: remove prop drilling

- CountrySection과 CategorySection의 데이터를 필요한 위치에서 useLoaderData로 받아옵니다.
- defaultOpenRegion을 useFeedStore에 저장하여 사용합니다.

seoulyego committed:
KL-162/rename: rename Response to Data

- feedLoader의 Response를 Data로 변경하였습니다.

seoulyego committed:
KL-162/test: disable AdditionalFilter

- AdditionalFilter의 변경 작업을 위해 FeedPage에서 비활성화하였습니다.

seoulyego commented:
AdditionalFilter의 fetch는 변경 작업 중입니다.
완료되면 추가 PR로 올리겠습니다.

seoulyego committed:
KL-162/refactor: separate initialize state

- state의 사전 설정을 useInitializeState로 분리하였습니다.

sayoonnn commented on src/hooks/useProductData.js:
```diff
@@ -1,59 +1,44 @@
-    ])
+    })
   }, [pageData, selectedQueryArray])
 
```
useEffect 사용하면 마운트 될 때 한번, selectedQueryArray 변경될 때 한번 해서 불필요하게 두 번 될 것 같습니다

sayoonnn commented on src/pages/FeedPage/components/BasicFilter/CategorySection.jsx:
```diff
@@ -89,7 +90,56 @@ CategoryCheckBox.propTypes = {
+        })
+      }
+    }
+  }, [isLoading, isError, data])
```
이렇게 하면 3번 호출되지 않나요?
마운트 될 때, isLoading 변경되었을 때, data 변경되었을 때
isLoading만 의존성으로 걸고 useEffect 안에서 isError와 data를 확인하는 게 낫지 않나요

seoulyego replied:
이후 커밋에서 loader의 적용으로 사라진 부분입니다.

sayoonnn reviewed (commented):
comment 달린 부분만 확인해주세요

seoulyego committed:
KL-162/refactor: apply new API for AdditionalFilter

- 기존의 Tag Data를 얻기 위한 fetch를 제거하였습니다.
- Categories Data로부터 Tag Data를 가져옵니다.
- TagSwitchList를 TagSwitchArray로 이름을 수정하였습니다.
- FeedPage의 AdditionalFilter를 활성화하였습니다.

seoulyego committed:
KL-162/refactor: rename subCategory

- subCategory 이름을 subcategory로 변경하였습니다.

seoulyego committed:
KL-162/rename: rename SubCategoryArray

- SubCategoryArray.jsx -> SubcategoryArray.jsx

seoulyego committed:
KL-162/style: change the debounce time

- debounce 시간 조정

sayoonnn commented on src/hooks/useTagData.js:
```diff
@@ -0,0 +1,39 @@
+
+const useTagData = () => {
+  const { categoryData: data } = useLoaderData()
+  const { selectedCategory, selectedSubcategory } = useFeedStore()
```
이렇게 사용하면 국가가 바뀌어도 다시 렌더링 될 거예요
state 사용해서 일부만 구독하는 방식으로 바꾸는게 좋을 것 같습니다

sayoonnn reviewed (dismissed):
state 변경 되지 않는 문제는 나중에 해결하는걸로 합니다

seoulyego committed:
KL-162/fix: use shallow store states

- store의 state를 지정해주었습니다.
- useShallow를 적용하였습니다.
